torch/_inductor/codegen/cpp.py

```diff
@@ -1994,10 +2012,39 @@ def __init__(self, kernel_group):
 
     def legalize_bf16(self, nodes):
         def add_to_dtype(sub_graph: torch.fx.Graph):
+            def only_used_by_store(node: torch.fx.Node):
+                if len(node.users) > 1:
+                    return False
+                for user in node.users:
+                    _only_used_by_store = user.target == "store"
+                    opt_ctx = OptimizationContext()
+                    opt_ctx.is_directly_load_bf16 = _only_used_by_store
+                    node.meta[OptimizationContext.key] = opt_ctx
+                    return _only_used_by_store
+
+            def can_directly_store(node: torch.fx.Node):
+                store_buf = node.args[1]
+                store_node = node.args[3]
+                store_dtype = V.graph.get_dtype(store_buf)
+                load_dtype = (
+                    V.graph.get_dtype(store_node.args[1])
+                    if store_node.target == "load"
+                    else store_node.args[-1]
+                )
+                if only_used_by_store(store_node) and store_dtype == load_dtype:
+                    opt_ctx = OptimizationContext()
+                    opt_ctx.is_directly_store_bf16 = True
+                    node.meta[OptimizationContext.key] = opt_ctx
+                    return True
+                else:
+                    return False
```

Comment:
Could we combine the two functions as a single one? We just need to support the loop body is memcpy.

Reply:
Also, what's the difference between "is_directly_load_bf16" and "is_directly_store_bf16"? Do they serve the same purpose?

Reply:
Yes, the two functions and two flags can be combined as a single one